### DISCUSSION TITLE
How to deploy VaultWarden at an enterprise scale?

### DISCUSSION THREAD

**Discussion by EthanAmvt** [dani-garcia/vaultwarden#4580] [Q&A]
I am an intern, and I need to create a new free password manager using VaultWarden.
I have managed to create a functional interface with Docker. However, it is not optimized to work with hundreds of users. I know that some companies have successfully developed it. Do you have any tutorials or documentation to address this issue? (Im not english so sorry for bad description)

**Comment by stefan0xC** [comment]
> However, it is not optimized to work with hundreds of users.

What do you mean?

**Reply by EthanAmvt** [reply]
I would like to automate invitations and automate the assignment of groups to each user, using a database for example.
each employee has an email address like this [last name]@[private-domaine.fr]
I would like to make sure that they all have an account and that their account has access to certain password databases.
I'm actively looking for a way to do this or an alternative. I read that it wasn't possible but there are examples of companies that have done it

**Reply by BlackDex** [reply]
That is where the [Directory Connector](https://bitwarden.com/help/directory-sync/) is used for together with an LDAP/AD environment.
There you can link users to groups and give specific users specific rights.

Vaultwarden supports the latest Directory Connector currently available [Version 2024.3.2](https://github.com/bitwarden/directory-connector/releases/tag/v2024.3.2)

**Reply by EthanAmvt** [reply]
Thank you for your response, but the solution you provided seems to work for Bitwarden, and I am using Vaultwarden. Do you have an alternative? Thanks.

**Reply by jobritz** [reply]
The Directory Connector from Bitwarden is compatible with Vaultwarden, you can use this connector with Vaultwarden as well

**Reply by BlackDex** [reply]
As i stated in my message.

**Reply by EthanAmvt** [reply]
my bad, I miss this point (it is write vaultwarden only on the official website too ^^')
Do you know where can I find a tutorial for installing it ?

**Reply by jobritz** [reply]
https://youtu.be/GmSRVsQMz9E?si=3bk7qS_Q6DnufAlO The Directory Connector Setup is explained at minute 8:30. You might need different directory Settings if you use another type of directory. For automisation I would recommend to take a look here: https://bitwarden.com/help/directory-sync/
